modules/apps/transfer/keeper/relay.go

```diff
@@ -230,6 +236,13 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 			return errorsmod.Wrap(err, "unable to unescrow tokens, this may be caused by a malicious counterparty module or a bug: please open an issue on counterparty module")
 		}
 
+		// get the existing total amount in escrow
+		currentTotalEscrow := k.GetTotalEscrowForDenom(ctx, token.GetDenom())
+		newTotalEscrow := currentTotalEscrow.Sub(token.Amount)
+
+		// store the new total amount in escrow
+		k.SetTotalEscrowForDenom(ctx, token.GetDenom(), newTotalEscrow)
```

Comment:
In a followup pr, we could consider bundling this with the bank send into two helper functions:
```go
    // escrow token case
    
    escrowAddress := types.GetEscrowAddress(...)
    if err := escrowToken(ctx, sender, escrowAddress, token); err != nil {
        return err
    }
    
    // unescrow case
    
    escrowAddress := types.GetEscrowAddress(...)
    if err := unescrowToken(ctx, escrowAddress, receiver, token); err != nil {
        return err
    }
    
    
    ... 
    
func escrowToken(ctx sdk.Context, sender, escrowAddress sdk.AccAddress, token sdk.Coin) {
        // escrow source tokens. It fails if balance insufficient.
	if err := k.bankKeeper.SendCoins(
		ctx, sender, escrowAddress, sdk.NewCoins(token),
	); err != nil {
		return err
	}

	// get the existing total amount in escrow
	currentTotalEscrow := k.GetTotalEscrowForDenom(ctx, token.GetDenom())
	newTotalEscrow := currentTotalEscrow.Add(token.Amount)

	// store the new total amount in escrow
	k.SetTotalEscrowForDenom(ctx, token.GetDenom(), newTotalEscrow)
		
	return nil
}
    

```

Reply:
Good idea! I will add it to the list of follow-ups in https://github.com/cosmos/ibc-go/issues/3368.